config/config.go

```diff
@@ -161,6 +166,177 @@ type AuthConfig struct {
 	AppendMetadataHeaders bool          `yaml:"append_metadata_headers,omitempty" json:"append_metadata_headers,omitempty"`
 }
 
+// EnvConfigs contains environment configs or URIs to them.
+type EnvConfigs struct {
+	// A list of strings containing environment config URIs
+	// Only the local file system is supported currently, e.g., file://path/to/config.yaml
+	ConfigURIs []string `yaml:"config_uris,omitempty" json:"config_uris,omitempty"`
+
+	// A list of environment configs
+	Inline []EnvironmentConfig `yaml:"inline,omitempty" json:"inline,omitempty"`
+}
+
+// EnvironmentConfig is an Apigee Environment-level config for
+// Envoy Adapter. It contains a list of operations for the adapter to
+// perform request authentication and authorization.
+type EnvironmentConfig struct {
+	// Unique ID of the environment config
+	ID string `yaml:"id" json:"id"`
+
+	// A list of proxy configs
+	ProxyConfigs []ProxyConfig
+}
+
+// ProxyConfig has the proxy configuration.
+type ProxyConfig struct {
+	// Top-level basepath for the proxy config
+	Basepath string `yaml:"basepath,omitempty" json:"basepath,omitempty"`
+
+	// A list of Operations, names of which must be unique within the proxy config.
+	Operations []APIOperation `yaml:"operations,omitempty" json:"operations,omitempty"`
+}
+
+// An APIOperation associates a set of rules with a set of request matching
+// settings.
+type APIOperation struct {
+	// Name of the operation. Unique within a APIRuntimeControlConfig.
+	Name string `yaml:"name" json:"name"`
+
+	// Authentication defines the AuthenticationRequirement
+	Authentication AuthenticationRequirement `yaml:"authentication,omitempty" json:"authentication,omitempty"`
+
+	// ConsumerAuthorization defines the ConsumerAuthorization
+	ConsumerAuthorization ConsumerAuthorization `yaml:"consumer_authorization,omitempty" json:"consumer_authorization,omitempty"`
+
+	// HTTP matching rules for this operation.
+	HTTPMatches []HTTPMatch `yaml:"http_match,omitempty" json:"http_match,omitempty"`
+
+	// Name of the target server for this operation. This will be sent to Envoy
+	// for routing to the corresponding upstream cluster upon a successful
+	// authorization of the operation.
+	Target string `yaml:"target" json:"target"`
+}
+
+// AuthenticationRequirement defines the authentication requirement.
+// Precisely one of JWT, Any and All should be set.
```

Comment:
Should we also convert this to an interface here or keep it as it and validate it ourselves? @josheinhorn @theganyo